packages/lesswrong/lib/hocUtils.js

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+export const hookToHoc = (hookFn) => {
```

Comment:
Have this function take a `propsToArguments` function, which allows you to control how the hook will be called